### ISSUE TITLE
Some components do not work in version 4.1.0?

### ISSUE THREAD

**Issue by be90728** [ComfortablyCoding/strapi-plugin-transformer#12]
In strapi version 4.0.7, getting response like below
```
{
  "data": {
    "id": 1,
    "title": "the title",
    "componentA": {
        "id": 1,
        "content": "Hello!",
    }
    "createdAt": "2022-02-11T01:51:49.902Z",
    "updatedAt": "2022-02-11T01:51:52.797Z",
    "publishedAt": "2022-02-11T01:51:52.794Z",
  },
  "meta": {},
}
```

But when I upgrade strapi to version 4.1.0, the response I get is like below
```
{
  "data": {
    "id": 1,
    "title": "the title",
    "componentA": {
        "data": {           <--- here
            "id": 1,
            "content": "Hello!",
        }
    }
    "createdAt": "2022-02-11T01:51:49.902Z",
    "updatedAt": "2022-02-11T01:51:52.797Z",
    "publishedAt": "2022-02-11T01:51:52.794Z",
  },
  "meta": {},
}
```

Strapi version: 4.1.0
Plugin version: 1.0.3

**Comment by ComfortablyCoding** [comment]
Hello @be90728 ,

Thank you for the report. I cannot reproduce this issue, can you please include a reproducable example?

With this schema
```json
{
  "kind": "collectionType",
  "collectionName": "test_12",
  "info": {
    "singularName": "test-12",
    "pluralName": "test-12",
    "displayName": "test-12",
    "description": ""
  },
  "options": {
    "draftAndPublish": true
  },
  "pluginOptions": {},
  "attributes": {
    "title": {
      "type": "string"
    },
    "componentA": {
      "type": "component",
      "repeatable": false,
      "component": "single.compo"
    }
  }
}
```

I receive this response for both 4.0.7 and 4.1.0 with the plugin enabled.
```json
{
  "data": [
    {
      "id": 1,
      "title": "the title",
      "createdAt": "2022-02-20T00:40:21.635Z",
      "updatedAt": "2022-02-20T00:40:23.300Z",
      "publishedAt": "2022-02-20T00:40:23.297Z",
      "componentA": {
        "id": 1,
        "content": "hello!"
      }
    }
  ],
  "meta": {}
}
```

**Comment by ComfortablyCoding** [comment]
Closing this issue for now, if a reproducible example is provided we can reopen/investigate.

**Comment by adriaanh** [comment]
I have the same problem, but seems to be related to https://github.com/strapi/strapi/pull/12579

**Comment by adriaanh** [comment]
With the following content type
```json
{
  "kind": "collectionType",
  "collectionName": "beers",
  "info": {
    "singularName": "beer",
    "pluralName": "beers",
    "displayName": "Beer",
    "description": ""
  },
  "options": {
    "draftAndPublish": false
  },
  "pluginOptions": {},
  "attributes": {
    "name": {
      "type": "string"
    },
    "bid": {
      "type": "integer"
    },
    "description": {
      "type": "text"
    },
    "label": {
      "type": "string"
    },
    "style": {
      "type": "string"
    },
    "abv": {
      "type": "decimal"
    },
    "ibu": {
      "type": "integer"
    },
    "rating_score": {
      "type": "decimal"
    },
    "rating_count": {
      "type": "integer"
    },
    "brewery": {
      "type": "relation",
      "relation": "manyToOne",
      "target": "api::brewery.brewery",
      "inversedBy": "beers"
    },
    "userbeers": {
      "type": "relation",
      "relation": "oneToMany",
      "target": "api::userbeer.userbeer",
      "mappedBy": "beer"
    }
  }
}
``` 
I get the following response without population (data truncated):
```json
{
   "data":[
      {
         "id":11,
         "name":"TestBeer",
         "createdAt":"2022-02-24T20:23:10.693Z",
         "updatedAt":"2022-02-24T20:23:32.419Z",
         "bid":1,
         "description":"Test",
         "label":"test",
         "style":"test",
         "abv":1,
         "ibu":1,
         "rating_score":1,
         "rating_count":1
      }
   ],
   "meta":{
      "pagination":{
         "page":1,
         "pageSize":25,
         "pageCount":1,
         "total":1
      }
   }
}
```

When populating the ' brewery', I get the following response:
```json
{
   "data":[
      {
         "id":11,
         "name":"TestBeer",
         "createdAt":"2022-02-24T20:23:10.693Z",
         "updatedAt":"2022-02-24T20:23:32.419Z",
         "bid":1,
         "description":"Test",
         "label":"test",
         "style":"test",
         "abv":1,
         "ibu":1,
         "rating_score":1,
         "rating_count":1,
         "brewery":{
            "data":{                                <--  here
               "id":16,
               "name":"TestBrewery",
               "createdAt":"2022-02-24T20:23:28.161Z",
               "updatedAt":"2022-02-24T20:23:28.161Z",
               "label":"test",
               "country":"test",
               "brid":1,
               "beer_count":1,
               "type":"test",
               "description":"1"
            }
         }
      }
   ],
   "meta":{
      "pagination":{
         "page":1,
         "pageSize":25,
         "pageCount":1,
         "total":7
      }
   }
}
```
I use the following versions:
```json
  "dependencies": {
    "@strapi/plugin-i18n": "4.1.1",
    "@strapi/plugin-users-permissions": "4.1.1",
    "@strapi/strapi": "4.1.1",
    "sqlite3": "5.0.2",
    "strapi-plugin-transformer": "^1.0.3"
  }
```

**Comment by ComfortablyCoding** [comment]
@adriaanh, that is the expected response format. 

Relations are structured like so by default
```json
{
  "data": {
    "id": 1,
    "attributes": {
      "title": "lorem ipsum",
      "slug": "lorem-ipsum",
      "createdAt": "2022-02-23T04:28:55.264Z",
      "updatedAt": "2022-02-25T04:38:08.396Z",
      "publishedAt": "2022-02-25T04:38:08.393Z",
      "ipsum": {
        "data": {
          "id": 15,
          "attributes": {
            "title": "dolor sat",
            "slug": "dolor-sat",
            "createdAt": "2022-02-25T01:27:02.579Z",
            "updatedAt": "2022-02-25T04:37:10.310Z",
            "publishedAt": "2022-02-25T04:37:10.308Z"
          }
        }
      }
    }
  },
  "meta": {}
}
```

therefor removing the attributes key the expected (and current) output is as follows
```json
{
  "data": {
    "id": 1,
    "title": "lorem ipsum",
    "slug": "lorem-ipsum",
    "createdAt": "2022-02-23T04:28:55.264Z",
    "updatedAt": "2022-02-25T04:38:08.396Z",
    "publishedAt": "2022-02-25T04:38:08.393Z",
    "ipsum": {
      "data": {
        "id": 15,
        "title": "dolor sat",
        "slug": "dolor-sat",
        "createdAt": "2022-02-25T01:27:02.579Z",
        "updatedAt": "2022-02-25T04:37:10.310Z",
        "publishedAt": "2022-02-25T04:37:10.308Z"
      }
    }
  },
  "meta": {}
}
```

**Comment by adriaanh** [comment]
Hm in my opinion this was not in a previous version, but I will check. Thanks for your response.

**Comment by adriaanh** [comment]
I did just test with the same configuration as above, but different Strapi & Strapi plugin transformer versions:

```json
  "dependencies": {
    "@strapi/plugin-i18n": "4.0.7",
    "@strapi/plugin-users-permissions": "4.0.7",
    "@strapi/strapi": "4.0.7",
    "sqlite3": "5.0.2",
    "strapi-plugin-transformer": "^1.0.1"
  },
```

I get the following response, no data here:
```json
{
    "id": 9,
    "name": "Test beer",
    "createdAt": "2022-02-25T15:35:45.040Z",
    "updatedAt": "2022-02-25T15:35:45.040Z",
    "bid": 1,
    "description": "test",
    "label": "test",
    "style": "test",
    "abv": 1,
    "ibu": 1,
    "rating_score": 1,
    "rating_count": 1,
    "brewery": {
      "id": 5,
      "name": "Test brewery",
      "label": "test",
      "country": "test",
      "beer_count": 1,
      "brid": 1,
      "type": "test",
      "description": "test",
      "createdAt": "2022-02-25T15:36:01.014Z",
      "updatedAt": "2022-02-25T15:36:01.014Z"
    }
  }
```

After updating only the plugin to 1.0.3 I get the following response (I did not update the strapi version):

```json
  {
    "id": 9,
    "name": "Test beer",
    "createdAt": "2022-02-25T15:35:45.040Z",
    "updatedAt": "2022-02-25T15:35:45.040Z",
    "bid": 1,
    "description": "test",
    "label": "test",
    "style": "test",
    "abv": 1,
    "ibu": 1,
    "rating_score": 1,
    "rating_count": 1,
    "brewery": {
      "data": { <-- here
        "id": 5,
        "name": "Test brewery",
        "label": "test",
        "country": "test",
        "beer_count": 1,
        "brid": 1,
        "type": "test",
        "description": "test",
        "createdAt": "2022-02-25T15:36:01.014Z",
        "updatedAt": "2022-02-25T15:36:01.014Z"
      }
    }
  }
```

Seems to me there is a change in how this plugin works from 1.0.1 to 1.0.3?

**Comment by ComfortablyCoding** [comment]
@adriaanh as mentioned in #14, yes the behaviour did change between those versions. A bug in the transform algorithm that was fixed in #7 removed the data key as part of removeAttributes transform which it should not have. 

I am leaving this closed as the original report was for a difference in strapi version not the plugins versions. 

The behaviour for the data key being added/removed will be held in #14
